pkgs/service/api.go

```diff
@@ -416,6 +415,94 @@ func handleEligibleNodesCount(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// handleEligibleNodesCount godoc
+// @Summary Get eligible nodes count for a specific day
+// @Description Retrieves the total count of eligible nodes and optionally their corresponding slotIDs (controlled by the includeSlotDetails query param) for a specified data market address and day
+// @Tags Eligible Nodes
+// @Accept json
+// @Produce json
+// @Param includeSlotDetails query bool false "Set to true to include slotIDs in the response"
+// @Param request body EligibleNodesCountRequest true "Eligible nodes count payload"
+// @Success 200 {object} Response[EligibleNodes]
+// @Failure 400 {string} string "Bad Request: Invalid input parameters (e.g., day < 1 or day > current day, invalid data market address)"
+// @Failure 401 {string} string "Unauthorized: Incorrect token"
+// @Router /eligibleNodesCount [post]
+func handleEligibleNodesCount(w http.ResponseWriter, r *http.Request) {
+	var request EligibleNodesCountRequest
+	if err := json.NewDecoder(r.Body).Decode(&request); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	if request.Token != config.SettingsObj.AuthReadToken {
+		http.Error(w, "Incorrect Token!", http.StatusUnauthorized)
+		return
+	}
+
+	isValid := false
+	for _, dataMarketAddress := range config.SettingsObj.DataMarketAddresses {
+		if request.DataMarketAddress == dataMarketAddress {
+			isValid = true
+			break
+		}
+	}
+
+	if !isValid {
+		http.Error(w, "Invalid Data Market Address!", http.StatusBadRequest)
+		return
+	}
+
+	day, err := prost.FetchCurrentDay(common.HexToAddress(request.DataMarketAddress))
+	if err != nil {
+		http.Error(w, "Failed to fetch current day", http.StatusBadRequest)
+		return
+	}
+	currentDay := new(big.Int).Set(day)
```

Comment:
this is already cached by protocol state cacher and also by other sequencer components multiple times